### ISSUE TITLE
2019-12-01 tc018 M4SS the estimated concentration dataset includes an unexpected value of CEST_EXT for TIME==0.

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#74]
With commit 6e99fab and tc018 M4SS there's a value in the estimated concentation dataset at TIME==0 for CEST_EXT. This is not expected. Based upon 1) all FLGEXKEL==1 (no time points included in a terminal elimination phase from the data) 2) OPTIMIZEKEL==0 (no optimization), 3) Partial AUCs are all at design/sample collection points Intervals collected are 0-12 (midpt 6) and 12-24 (midpt 18 hours), it's expected that the estimated concentration dataset would have no records generated.

```
> mct[, names(parameter_required("^AUC.[0-9]+?.", names(mct), simplify=FALSE))]
  AUC.1.T1 AUC.1.T2 AUC.2.T1 AUC.2.T2
1        0       12        6       18
> table(d$PKPTMS)

 0 12 
45 45 
> table(d$PKPTME)

12 24 
45 45 
```

**Comment by rudraya** [comment]
I need some clarification on this, so currently the AURCT1_T2 is generated by taking the concentration (in this case, "rate") and time (in this case "midpoint") into consideration and use the points mentioned by AUC.X.T1 and AUC.X.T2 to get the Partial AUC. So for example, AUC.1.T1 is 0 and AUC.2.T2 is 12, the computation engine will try to compute partial AUC from 0 to 12 even though the midpoint data is reflected to be 6. Hence all the values for partial AUC for AUC0_12 will be NAs (this is the current behavior as of last commit). Please let me know if this is not what the intention of partial AUCs for M4 models.

**Comment by tensfeldt** [comment]
Everything mentioned in your comment is correct. 

What is unexpected is that while there is a value for AURC0_12 of all "NA" in the generated parameters dataset, 

```
> r
       SDEID AMT.12.00 AMT.24.00 AE.12.00 AE.24.00 AEPCT.12.00 AEPCT.24.00       AE    AEPCT   AETAU1  AETAUPT1  MAXRATE TMAXRATE MAXRATE1 TMAXRATE1 RATELAST1 MIDPTLAST1 TLAG KEL KELTMLO KELTMHI
1  102652097   79.3760   27.7380  79.3760 107.1140    39688000    53557000 107.1140 53557000 186.4900  93245000 6614.667        6 6614.667         6  2311.500         18    6  NA      NA      NA
2  155315252   41.3270   22.2250  41.3270  63.5520    20663500    31776000  63.5520 31776000 104.8790  52439500 3443.917        6 3443.917         6  1852.083         18    6  NA      NA      NA
3  212128320   66.4200   35.6890  66.4200 102.1090    33210000    51054500 102.1090 51054500 168.5290  84264500 5535.000        6 5535.000         6  2974.083         18    6  NA      NA      NA
4  232007206   43.4750   34.0375  43.4750  77.5125    43475000    77512500  77.5125 77512500 120.9875 120987500 3622.917        6 3622.917         6  3622.917          6    6  NA      NA      NA
5  303069272    4.3056    3.0774   4.3056   7.3830    14352000    24610000   7.3830 24610000  11.6886  38962000  358.800        6  358.800         6   256.450         18    6  NA      NA      NA
6  322184155   44.7300    4.6200  44.7300  49.3500    22365000    24675000  49.3500 24675000  94.0800  47040000 3727.500        6 3727.500         6   385.000         18    6  NA      NA      NA
7  346821548    8.5995    1.8480   8.5995  10.4475    28665000    34825000  10.4475 34825000  19.0470  63490000  716.625        6  716.625         6   154.000         18    6  NA      NA      NA
8  359622627   43.5400   34.5466  43.5400  78.0866    21770000    39043300  78.0866 39043300 121.6266  60813300 3628.333        6 3628.333         6  2878.883         18    6  NA      NA      NA
9  392430764   35.1200   28.1775  35.1200  63.2975    35120000    63297500  63.2975 63297500  98.4175  98417500 2926.667        6 2926.667         6  2926.667          6    6  NA      NA      NA
10 395827329   62.4800   28.0416  62.4800  90.5216    31240000    45260800  90.5216 45260800 153.0016  76500800 5206.667        6 5206.667         6  2336.800         18    6  NA      NA      NA
   KELNOPT KELR KELRSQ KELRSQA FLGACCEPTKEL THALF THALFF   AURCALL  AURCLAST     AURC1 AURCINT1 AURC0_12  AURC6_18 AURCINFO AURCINFP AURCXPCTO AURCXPCTP VOLSUM    RATE1    RATE2 MIDPT1 MIDPT2 DI1
1        0   NA     NA      NA            0    NA   <NA> 49113.902 49113.902 49113.902     6_18       NA 49113.902       NA       NA        NA        NA  3.760 6614.667 2311.500      6     18  NA
2        0   NA     NA      NA            0    NA   <NA> 30794.863 30794.863 30794.863     6_18       NA 30794.863       NA       NA        NA        NA  3.085 3443.917 1852.083      6     18  NA
3        0   NA     NA      NA            0    NA   <NA> 49473.908 49473.908 49473.908     6_18       NA 49473.908       NA       NA        NA        NA  2.940 5535.000 2974.083      6     18  NA
4        0   NA     NA      NA            0    NA   <NA> 38563.977 38563.977        NA     6_18       NA 38563.977       NA       NA        NA        NA  2.725 3622.917 2836.458      6     18  NA
5        0   NA     NA      NA            0    NA   <NA>  3657.192  3657.192  3657.192     6_18       NA  3657.192       NA       NA        NA        NA  2.160  358.800  256.450      6     18  NA
6        0   NA     NA      NA            0    NA   <NA> 17667.660 17667.660 17667.660     6_18       NA 17667.660       NA       NA        NA        NA  2.083 3727.500  385.000      6     18  NA
7        0   NA     NA      NA            0    NA   <NA>  4390.934  4390.934  4390.934     6_18       NA  4390.934       NA       NA        NA        NA  1.190  716.625  154.000      6     18  NA
8        0   NA     NA      NA            0    NA   <NA> 38870.054 38870.054 38870.054     6_18       NA 38870.054       NA       NA        NA        NA  2.166 3628.333 2878.883      6     18  NA
9        0   NA     NA      NA            0    NA   <NA> 31521.431 31521.431        NA     6_18       NA 31521.431       NA       NA        NA        NA  1.775 2926.667 2348.125      6     18  NA
10       0   NA     NA      NA            0    NA   <NA> 42985.811 42985.811 42985.811     6_18       NA 42985.811       NA       NA        NA        NA  1.608 5206.667 2336.800      6     18  NA

```
it was not anticipated that the estimated concentration dataset would have a value for CEST_EXT, nor am I sure I understand how it is generated.
```
> e
   PKDATAROWID     SDEID TIME CEST_KEL CEST_INT   CEST_EXT CEST_C0 CEST_TLAST
1           NA  43456658    0       NA       NA  527333.33      NA         NA
2           NA  98862581    0       NA       NA 2635416.67      NA         NA
3           NA 102652097    0       NA       NA 6614666.67      NA         NA
4           NA 155315252    0       NA       NA 3443916.67      NA         NA
5           NA 212128320    0       NA       NA 5535000.00      NA         NA
6           NA 232007206    0       NA       NA 3622916.67      NA         NA
7           NA 303069272    0       NA       NA  358800.00      NA         NA
8           NA 322184155    0       NA       NA 3727500.00      NA         NA
9           NA 346821548    0       NA       NA  716625.00      NA         NA
10          NA 359622627    0       NA       NA 3628333.33      NA         NA
11          NA 392430764    0       NA       NA 2926666.67      NA         NA
12          NA 395827329    0       NA       NA 5206666.67      NA         NA
13          NA 436516214    0       NA       NA 2867500.00      NA         NA
14          NA 446296844    0       NA       NA  582000.00      NA         NA
15          NA 448488588    0       NA       NA 5861250.00      NA         NA
16          NA 499736026    0       NA       NA  515533.33      NA         NA
17          NA 514418530    0       NA       NA 3123750.00      NA         NA
18          NA 522999139    0       NA       NA 2338583.33      NA         NA
19          NA 553304375    0       NA       NA 5304000.00      NA         NA
20          NA 578457836    0       NA       NA 4904166.67      NA         NA
21          NA 581080171    0       NA       NA 2880000.00      NA         NA
22          NA 582170099    0       NA       NA 4750016.67      NA         NA
23          NA 583844675    0       NA       NA  873070.83      NA         NA
24          NA 590276539    0       NA       NA 1113833.33      NA         NA
25          NA 598462265    0       NA       NA 5644166.67      NA         NA
26          NA 658062633    0       NA       NA 3785833.33      NA         NA
27          NA 678253176    0       NA       NA 2539125.00      NA         NA
28          NA 702151038    0       NA       NA 1572083.33      NA         NA
29          NA 722709781    0       NA       NA 3850500.00      NA         NA
30          NA 724829716    0       NA       NA  745875.00      NA         NA
31          NA 763781430    0       NA       NA 3960833.33      NA         NA
32          NA 774566237    0       NA       NA  849658.33      NA         NA
33          NA 813912334    0       NA       NA 4910208.33      NA         NA
34          NA 846570468    0       NA       NA  866858.33      NA         NA
35          NA 850772340    0       NA       NA 1011041.67      NA         NA
36          NA 869199587    0       NA       NA 5934000.00      NA         NA
37          NA 882441025    0       NA       NA 3109791.67      NA         NA
38          NA 889990267    0       NA       NA 3529166.67      NA         NA
39          NA 904183466    0       NA       NA 3607500.00      NA         NA
40          NA 909697662    0       NA       NA 3371250.00      NA         NA
41          NA 927550096    0       NA       NA 4745000.00      NA         NA
42          NA 934267135    0       NA       NA  641133.33      NA         NA
43          NA 936232252    0       NA       NA 1592500.00      NA         NA
44          NA 974609637    0       NA       NA   53453.75      NA         NA
45          NA 995384974    0       NA       NA 1395000.00      NA         NA
> 
```

**Comment by rudraya** [comment]
This issue was addressed in bb8e82d, please let me know if it still occurs.

**Comment by tensfeldt** [comment]
This was retested with commit bb8e82d and is resolved and will be closed.
